crates/biome_js_analyze/src/lint/nursery/no_bitwise.rs

```diff
@@ -0,0 +1,191 @@
+        }
+    }
+    fn exist_bitwise_op(&self) -> bool {
+        match self {
+            AnyExpressionWithBitwise::JsBinaryExpression(_) => {
+                self.operator_token().is_some_and(|op| {
+                    matches!(op.kind(), T![&] | T![|] | T![^] | T![<<] | T![>>] | T![>>>])
+                })
+            }
+            AnyExpressionWithBitwise::JsUnaryExpression(_) => self
+                .operator_token()
+                .is_some_and(|op| matches!(op.kind(), T![~])),
+            AnyExpressionWithBitwise::JsAssignmentExpression(_) => {
+                self.operator_token().is_some_and(|op| {
+                    matches!(
+                        op.kind(),
+                        T![&=] | T![|=] | T![^=] | T![<<=] | T![>>=] | T![>>>=]
+                    )
+                })
+            }
+        }
+    }
+}
+
+impl Rule for NoBitwise {
+    type Query = Ast<AnyExpressionWithBitwise>;
+    type State = JsSyntaxToken;
+    type Signals = Option<Self::State>;
+    type Options = NoBitwiseOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let expr = ctx.query();
+        if expr.exist_bitwise_op() {
+            let op = expr.operator_token().unwrap();
+            if ctx
+                .options()
+                .allow
+                .iter()
+                .any(|op_str| op_str.as_ref() == op.text_trimmed())
+            {
+                return None;
+            }
+            // check if the operator is '|' and right is '0'
+            if ctx.options().int32_hint && op.kind() == T![|] {
+                if let AnyExpressionWithBitwise::JsBinaryExpression(node) = expr {
+                    let right = node.right().ok()?.omit_parentheses();
+                    if op.kind() == T![|]
+                        && right
+                            .as_any_js_literal_expression()
+                            .is_some_and(|literal_expr| {
+                                literal_expr.as_js_number_literal_expression().is_some_and(
+                                    |number_literial| {
+                                        number_literial
+                                            .value_token()
+                                            .ok()
+                                            .is_some_and(|value| value.text_trimmed() == "0")
+                                    },
+                                )
+                            })
+                    {
+                        return None;
+                    }
+                }
+            }
+            return Some(op);
+        }
+        None
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let op = state.text_trimmed();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
```

Comment:
Our diagnostics follow the [rule pillars](https://biomejs.dev/linter/#rule-pillars): https://github.com/biomejs/biome/blob/21dd358be5dade01d672916a0ab76bc825ef85cb/crates/biome_analyze/CONTRIBUTING.md#guidelines

In this case, we don't tell the user what to do to fix the issue. We should provide a third note that does so.